content/how-to/develop.md

```diff
@@ -4,42 +4,146 @@ contributors:
 
-W> Explain that IT SHOULD NOT BE USED IN PRODUCTION, NEVER.
+webpack can be used with **watch mode**. In this mode webpack will watch your files, and recompile when they change.
+**webpack-dev-server** provides an easy to use development server with fast live reloading. If you already have a development server and want full flexibility, **webpack-dev-middleware** can be used as middleware.
 
-Adjusting your text editor
+webpack-dev-server and webpack-dev-middleware use in-memory compilation, meaning that the bundle will not be saved to disk. This makes compiling faster and results in less mess on your file system.
 
-- Explain that some editors use "atomic save"; this needs to be disabled in order to see all file changes.
+In most cases **you'll want to use webpack-dev-server**.
 
-Choose a devtool
+We will explain how to get started with developing with each of the tools described above.
 
-- Explain shortly what source maps are and why they come in handy when devving
-- Link to devtool page for more info about source maps
+W> Never use any of this in production. Ever.
 
-webpack watch mode
 
-- show one use case
-- show how to use in CLI
-- show how to use API
-- explain that you need to use your own server (maybe show the python server)
+### Adjusting Your Text Editor
 
-webpack-dev-server
+Some text editors have a "safe write" feature and enable this by default. As a result, saving a file will not always result in a recompile.
 
-- show one use case
-- how to install
-- show how to use in CLI
-- show how to use API
-- refer to `devServer` configuration page
-- explain `inline` and `iframe` mode, and explain `inline` requires more config in API
-- make clear that there are some differences in CLI and API
-- after explaining live reload, link to HMR article
+Each editor has a different way of disabling this. For the most common ones:
 
-webpack-dev-middleware
+* **Sublime Text 3** - Add `"atomic_save": false` to your user preferences.
+* **IntelliJ** - use search in the preferences to find "safe write" and disable it.
+* **Vim** - add `:set backupcopy=yes` in your settings.
 
-- show one use case
-- how to install
-- show how to use API
-- refer to `devServer` configuration page
 
+### Source Maps
+
+When a Javascript exception occurs, you'll often want to know what file and line is generating this error. Since webpack outputs files into one or more bundles, it can be inconvenient to trace the file.
+
+**Source maps** intend to fix this problem. There are a lot of [different options](../configuration/devtool) - each with their own advantages and disadvantages. To get started, we'll use this one:
+
+```
+devtool: "cheap-eval-source-map"
+```
+
+
+### webpack Watch Mode
+
+webpack's watch mode watches files for changes. If any change is detected, it'll run the compilation again.
+
+We want webpack to display nice colors and a progress bar while it's compiling. Let's run the command:
```

Comment:
If I remember right, `--colors` shouldn't be needed anymore. webpack should be able to detect color support automatically. The flag might not exist. Check this.